src/transformation/visitors/modules/export.ts

```diff
@@ -46,24 +47,62 @@ function transformExportAllFrom(context: TransformationContext, node: ts.ExportD
     const moduleRequire = createModuleRequire(context, node.moduleSpecifier);
     const tempModuleIdentifier = lua.createIdentifier("____export");
 
+    const result: lua.Statement[] = [];
+
     const declaration = lua.createVariableDeclarationStatement(tempModuleIdentifier, moduleRequire);
+    result.push(declaration);
+
+    if (node.exportClause && ts.isNamespaceExport(node.exportClause)) {
```

Comment:
Could you add a comment explaining in what case this if statement is needed and what this branch of this function does?